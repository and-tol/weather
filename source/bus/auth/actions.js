// Types
import { types } from './types';

export const authActions = {

    // Sync
    authenticate: () => {
        return {
            type: types.AUTHENTICATE,
        };
    },

    // Async

    loginAsync: (userData) => {

        return {
            type:    types.LOGIN_ASYNC,
            payload: userData,
        };
    },

    resetPasswordAsync: (login) => {
        return {
            type:    types.RESET_PASSWORD_ASYNC,
            payload: login,

        };
    },

};
