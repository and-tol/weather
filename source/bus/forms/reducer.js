// Core
import { combineForms } from 'react-redux-form';
export const formsReducer = combineForms({
    user: {
        profile: {
            login:    '',
            password: '',
        },
    },
},
'forms'
);
