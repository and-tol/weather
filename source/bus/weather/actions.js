// Types
import { types } from './types';

export const weatherActions = {
    // Sync
    fillWeather: (weatherData) => {
        return {
            type:    types.FILL_WEATHER,
            payload: weatherData,
        };
    },

    // Async

    getWeather: (idCities) => {
        return {
            type:    types.GET_WEATHER,
            payload: idCities,
        };
    },
};
