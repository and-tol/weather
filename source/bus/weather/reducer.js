// Types
import { types } from './types';
const initialState = {
    cities: [
        {
            name: 'Владивосток',
            link: 'vladivostok',
            id:   2013348,
        },
        {
            name: 'Сочи',
            link: 'sochi',
            id:   491422,
        },
        {
            name: 'Ярославль',
            link: 'yaroslavl',
            id:   468902,
        }
    ],
    weatherData: [],
};

export const weatherReducer = (state = initialState, action) => {
    switch (action.type) {
        case types.FILL_WEATHER:
            state.weatherData = action.payload;

            return {
                ...state,
            };

        default:
            return state;
    }
};
