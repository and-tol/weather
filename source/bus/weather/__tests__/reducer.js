// Reducer
import { weatherReducer } from '../reducer';

// Actions
import { weatherActions } from '../actions';

const initialState = {
    cities: [
        {
            name: 'Владивосток',
            link: 'vladivostok',
            id:   2013348,
        },
        {
            name: 'Сочи',
            link: 'sochi',
            id:   491422,
        },
        {
            name: 'Ярославль',
            link: 'yaroslavl',
            id:   468902,
        }
    ],
    weatherData: [],
};

describe('weather reducer:', () => {
    test('should return initial state by default', () => {
        expect(weatherReducer(void 0, {})).toEqual(initialState);
    });
    test('should handle FILL_WEATHER action', () => {
        expect(
            weatherReducer(void 0, weatherActions.fillWeather(__.weatherMock))
        ).toMatchSnapshot();
    });
});
