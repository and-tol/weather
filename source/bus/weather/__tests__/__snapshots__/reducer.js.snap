// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`weather reducer: should handle FILL_WEATHER action 1`] = `
Object {
  "cities": Array [
    Object {
      "id": 2013348,
      "link": "vladivostok",
      "name": "Владивосток",
    },
    Object {
      "id": 491422,
      "link": "sochi",
      "name": "Сочи",
    },
    Object {
      "id": 468902,
      "link": "yaroslavl",
      "name": "Ярославль",
    },
  ],
  "weatherData": Array [
    Object {
      "clouds": Object {
        "all": 40,
      },
      "coord": Object {
        "lat": 43.11,
        "lon": 131.87,
      },
      "dt": 1556695234,
      "id": 2013348,
      "main": Object {
        "humidity": 76,
        "pressure": 1000,
        "temp": 10,
        "temp_max": 10,
        "temp_min": 10,
      },
      "name": "Vladivostok",
      "sys": Object {
        "country": "RU",
        "id": 8885,
        "message": 0.005,
        "sunrise": 1556654827,
        "sunset": 1556705525,
        "type": 1,
      },
      "visibility": 10000,
      "weather": Array [
        Object {
          "description": "слегка облачно",
          "icon": "03d",
          "id": 802,
          "main": "Clouds",
        },
      ],
      "wind": Object {
        "speed": 1,
      },
    },
  ],
}
`;
